### DISCUSSION TITLE
Get typesafe child routes list

### DISCUSSION THREAD

**Discussion by UltimateGG** [TanStack/router#2197] [Q&A] [answered]
I often find myself making functions that for reusability sake have shorter paths.
For example if I had
`/page/subpage/a`
`/page/subpage/b`
`/page/subpage/c/hello`

I may have a function, component, link, etc. like this:
```ts
const changeSubPage = (subPage: string) => {
    navigate({ to: '/page/subpage/' + subPage });
}
```

Is there a way to get that `subPage: string` to be `'/a' | '/b' | '/c/hello'`?
You can get the link type, but that is all full paths (#429). This may be tricky because its strings vs typescript but I thought it may be possible because in routegen file it does contain the separated paths (path, and fullPath)

**Comment by schiller-manuel** [comment, marked as answer]
You could use something like this:

```tsx
type Parent = '/page/subpage/'
type ParentRoute = RouteByPath<RegisteredRouter['routeTree'], Parent>
type ChildRoutes = ParentRoute['types']['children'][keyof ParentRoute['types']['children']]
type ChildPaths = ChildRoutes['path']

const changeSubPage = (subPage: ChildPaths) => {
      navigate({ to: `/page/subpage/${subPage}`});
}
```

**Reply by UltimateGG** [reply]
Awesome, thanks!
